### PR TITLE
Update web3 api to compute eth transfers to an address

### PR DESCRIPTION
**Pull request by harisang**
This PR adjusts a bit some of the functionality of the web3 api and adds a function that computes the native eth transfers to a target address. This again is needed for the PR that tracks the MEV Blocker kickbacks (that will be a follow-up to this PR).

### CODE REVIEW

File: src/apis/web3api.py [hunk 1]
```diff
@@ -45,35 +45,60 @@ def get_current_block_number(self) -> Optional[int]:
...
+
+        if log_receipts is None:
+            return None
         settlement_hashes_list = list(
             {log_receipt["transactionHash"].hex() for log_receipt in log_receipts}
         )
-
         return settlement_hashes_list
 
+    def get_incoming_eth_transfers_to_contract_within_block_range(
```
Comment (fhenneke) [inline src/apis/web3api.py:87]: The name is very descriptive but also looks a bit long. Something like `get_incoming_eth_transfers` or `get_eth_transfers_by_block` would also work for me and are a bit more consistent with the other function name `get_tx_hash_by_block`.

File: src/apis/web3api.py [hunk 2]
```diff
@@ -45,35 +45,60 @@ def get_current_block_number(self) -> Optional[int]:
...
+        self, start_block: int, end_block: int, target: str
```
Comment (fhenneke) [inline src/apis/web3api.py:88]: Is the target always the settlement contract? Or is it WETH address or something else?

File: src/apis/web3api.py [hunk 3]
```diff
@@ -45,35 +45,60 @@ def get_current_block_number(self) -> Optional[int]:
...
+        """
+        Function that computes total eth transfers to a target address
+        within a certain block range
+        """
+        log_receipts = self.get_filtered_receipts(start_block, end_block, target, [])
+        if log_receipts is None:
+            return None
+        total_transfers_in_eth = 0.0
+        for txs in log_receipts:
+            total_transfers_in_eth += int(txs["data"].hex(), 16) / 10**18
```
Comment (fhenneke) [inline src/apis/web3api.py:99]: I got an error here calling `web3api.get_incoming_eth_transfers_to_contract_within_block_range(18423905 - 5, 18423905 + 5, SETTLEMENT_CONTRACT_ADDRESS)`:
`OverflowError: integer division result too large for a float`
From the looks of it there are multiple transactions returned for the filter and not all of the transactions are handled correctly.

**Review by fhenneke — CHANGES_REQUESTED** [review]
The old functionality seems to be working fine locally.

I do not yet understand the new functionality.  Potentially there is still bug in there.

Could you give an example for how you imagine the code to be called?

File: src/apis/web3api.py (hunk 3 above)
Comment (harisang) [reply src/apis/web3api.py]: Added an `if` condition, inspired by this.
https://github.com/5afe/safe-indexer/issues/5

Now it should only check incoming eth transfers for Safe addresses.

File: src/apis/web3api.py (hunk 2 above)
Comment (harisang) [reply src/apis/web3api.py:89]: With the fix, the target is intended now to be any Safe address

File: src/apis/web3api.py (hunk 1 above)
Comment (harisang) [reply src/apis/web3api.py]: Just made it shorter and chose `get_eth_transfers_by_block_range`

File: src/apis/web3api.py (hunk 3 above)
Comment (harisang) [reply src/apis/web3api.py]: Here is an example, for some more context.

https://etherscan.io/tx/0xa0db246917a16cc0b2bd27fcddbe35d304c9563fba2087d1ee45650822e2a7f1#eventlog

**Comment by harisang** [comment]
@fhenneke Could you have a look again?

**Review by fhenneke — APPROVED** [review]
I did run the following command
```python
web3api.get_eth_transfers_by_block_range(18000000, 18400000, "0xCe91228789B57DEb45e66Ca10Ff648385fE7093b")
```
and the result was `79.41965354067895`.

If this seems reasonable, this is good to be merged.

**Comment by harisang** [comment]
> ```python
> 18000000
> ```

Here is a dune query that (almost....) matches this number. Not sure why there is a discrepancy, tbh.
https://dune.com/queries/3166955?start_block_t6c1ea=18000000&end_block_t6c1ea=18400000

**Comment by harisang** [comment]
> I did run the following command
> 
> ```python
> web3api.get_eth_transfers_by_block_range(18000000, 18400000, "0xCe91228789B57DEb45e66Ca10Ff648385fE7093b")
> ```
> 
> and the result was `79.41965354067895`.
> 
> If this seems reasonable, this is good to be merged.



Here is a dune query that (almost....) matches this number. Not sure why there is a discrepancy, tbh.
https://dune.com/queries/3166955?start_block_t6c1ea=18000000&end_block_t6c1ea=18400000

**Comment by fhenneke** [comment]
> Not sure why there is a discrepancy, tbh.
This makes me a bit nervous 😅

**Pull request merged**